app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutBatteryDevice.java

```diff
@@ -0,0 +1,181 @@
+         * }
+         */
+        @Override
+        public JSONObject getUploaderJson(Context mContext) throws JSONException {
+            JSONObject uploader = new JSONObject();
+
+            Ob1DexTransmitterBattery b = new Ob1DexTransmitterBattery();
+
+            if (!b.isPresent()) return null;
+
+            uploader.put("days", b.days());
+            uploader.put("daysEstimate", b.daysEstimate());
+            uploader.put("status", b.status().name());
+            uploader.put("voltagea", b.voltageA());
+            if (b.voltageAWarning()) {
+                uploader.put("voltagea_warning", true);
+            }
+            uploader.put("voltageb", b.voltageB());
+            if (b.voltageBWarning()) {
+                uploader.put("voltageb_warning", true);
+            }
+            uploader.put("resistance", b.resistance());
+            if (b.resistanceStatus() != Ob1DexTransmitterBattery.ResistanceStatus.UNKNOWN) {
+                uploader.put("resistance_status", b.resistanceStatus().name());
+            }
+            uploader.put("temperature", b.temperature());
+
+            // What nightscout will normally show in the UI
+            uploader.put("battery", b.days() + " days (voltage: " + b.voltageA() + "/" + b.voltageB() + ")");
+
+            uploader.put("type", name());
+            return uploader;
+        }
+
+        /**
+         * Only sends to Nightscout when there is a change in the voltage A value.
+         */
+        @Override
+        boolean alwaysSendBattery() {
+            return false;
+        }
+
+        @Override
+        String getDeviceName() {
+            if (Ob1G5StateMachine.usingG6()) {
+                return "G6 Transmitter";
+            }
+            return "G5 Transmitter";
+        }
+    };
+
+
+    /**
+     * Returns the battery level of the device.
+     *
+     * @param mContext is supplied so that the phone battery can be retrieved.
+     */
+    abstract int getBatteryLevel(Context mContext);
+
+    /**
+     * Returns the JSON object containing the battery details.
+     * Defaults to returning a JSONObject with integer key battery (required by Nightscout)
+     * and a string key type with the value of the enum (e.g. PHONE, DEXCOM_TRANSMITTER)
+     * @param mContext is supplied so that the phone battery can be retrieved.
+     */
+    public JSONObject getUploaderJson(Context mContext) throws JSONException {
+        JSONObject uploader = new JSONObject();
+        uploader.put("battery", getBatteryLevel(mContext));
+        uploader.put("type", name());
+        return uploader;
+    }
+
+    /**
+     * Returns the device name sent to Nightscout for the device.
+     */
+    abstract String getDeviceName();
+
+    /**
+     * If true, the battery status is always uploaded to Nightscout.
+     * If false, the battery status is only uploaded when it changes.
+     *
+     * Nightscout doesn't currently display device device status if it thinks its stale,
```

Comment:
remove the second `device` and replace `its` with `it's`.

Reply:
That's comment was copied from somewhere else, I know.

Reply:
Indeed it was!